default.nix

```diff
@@ -0,0 +1,277 @@
+        is given by 'mail.example.com'
+      '';
+    };
+
+    login_accounts = mkOption {
+      type = types.loaOf (types.submodule ({ name, ... }: {
+        options = {
+          name = mkOption {
+            type = types.str;
+            example = "user1";
+            description = "Username";
+          };
+
+          hashedPassword = mkOption {
+            type = types.str;
+            example = "$6$evQJs5CFQyPAW09S$Cn99Y8.QjZ2IBnSu4qf1vBxDRWkaIZWOtmu1Ddsm3.H3CFpeVc0JU4llIq8HQXgeatvYhh5O33eWG3TSpjzu6/";
+            description = ''
+              Hashed password. Use `mkpasswd` as follows
+
+              ```
+              mkpasswd -m sha-512 "super secret password"
+              ```
+            '';
+          };
+        };
+
+        config.name = mkDefault name;
+      }));
+      example = {
+        user1 = {
+          hashedPassword = "$6$evQJs5CFQyPAW09S$Cn99Y8.QjZ2IBnSu4qf1vBxDRWkaIZWOtmu1Ddsm3.H3CFpeVc0JU4llIq8HQXgeatvYhh5O33eWG3TSpjzu6/";
+        };
+        user2 = {
+          hashedPassword = "$6$oE0ZNv2n7Vk9gOf$9xcZWCCLGdMflIfuA0vR1Q1Xblw6RZqPrP94mEit2/81/7AKj2bqUai5yPyWE.QYPyv6wLMHZvjw3Rlg7yTCD/";
+        };
+      };
+      description = ''
+        The login account of the domain. Every account is mapped to a unix user,
+        e.g. `user1@example.com`. To generate the passwords use `mkpasswd` as
+        follows
+
+        ```
+        mkpasswd -m sha-512 "super secret password"
+        ```
+      '';
+    };
+
+    valiases = mkOption {
+      type = types.attrsOf (types.enum (builtins.attrNames cfg.login_accounts));
+      example = {
+        info = "user1";
+        postmaster = "user1";
+        abuse = "user1";
+      };
+      description = ''
+        Virtual Aliases. A virtual alias `info = "user1"` means that
+        all mail to `info@example.com` is forwarded to `user1@example.com`. Note
+        that it is expected that `postmaster@example.com` and `abuse@example.com` is
+        forwarded to some valid email address. (Alternatively you can create login
+        accounts for `postmaster` and (or) `abuse`).
+      '';
+    };
+
+    vmail_id_start = mkOption {
+      type = types.int;
+      default = 5000;
+      description = ''
+        The unix UID where the login_accounts are created. 5000 means that the first
+        user will get 5000, the second 5001, ...
+      '';
+    };
+
+    vmail_user_name = mkOption {
+      type = types.str;
+      default = "vmail";
+      description = ''
+        The user name and group name of the user that owns the directory where all
+        the mail is stored.
+      '';
+    };
+
+    vmail_group_name = mkOption {
+      type = types.str;
+      default = "vmail";
+      description = ''
+        The user name and group name of the user that owns the directory where all
+        the mail is stored.
+      '';
+    };
+
+    mail_dir = mkOption {
+      type = types.string;
```

Comment:
Shouldn't be this `types.path`?

Reply:
Ah yes, gonna correct it.

Also: README needs to be changed a bit (don't merge yet)